api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -141,44 +145,75 @@ public Cohort(String commaSeparatedIds) {
 	 */
 	public String getCommaSeparatedPatientIds() {
 		StringBuilder sb = new StringBuilder();
-		for (Iterator<Integer> i = getMemberIds().iterator(); i.hasNext();) {
-			sb.append(i.next());
-			if (i.hasNext()) {
-				sb.append(",");
-			}
+		for (CohortMembership member : getMembers()) {
+			sb.append(member.getPatient().getPatientId());
+			sb.append(",");
 		}
 		return sb.toString();
 	}
-	
-	public boolean contains(Patient patient) {
-		return getMemberIds() != null && getMemberIds().contains(patient.getPatientId());
-	}
-	
+
+	// TODO: Compare patient objects instead of IDs after removal of deprecated methods
 	public boolean contains(Integer patientId) {
-		return getMemberIds() != null && getMemberIds().contains(patientId);
+		return getMembers() != null && getMembers().stream()
+				.anyMatch(m -> m.getPatient().getPatientId().equals(patientId) && m.isMemberActive());
 	}
 	
+	public boolean contains(Patient patient) {
+		return contains(patient.getPatientId());
+	}
+
 	public String toString() {
 		StringBuilder sb = new StringBuilder("Cohort id=" + getCohortId());
 		if (getName() != null) {
-			sb.append(" name=" + getName());
+			sb.append(" name=").append(getName());
 		}
-		if (getMemberIds() != null) {
-			sb.append(" size=" + getMemberIds().size());
+		if (getMembers() != null) {
+			sb.append(" size=").append(getMembers().size());
 		}
 		return sb.toString();
 	}
-	
+
 	public void addMember(Integer memberId) {
-		getMemberIds().add(memberId);
+		this.addMembership(new CohortMembership(new Patient(memberId)));
 	}
-	
-	public void removeMember(Integer memberId) {
```

Comment:
The ticket seems to suggest that existing members should not change their signatures but simply delegate to the new ones. Did you intentionally get rid of removeMember which takes an integer member id?